### PR TITLE
Avoid leaking db transactions and rows

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

Addresses #553 and #517.

### CODE REVIEW

File: idb/postgres/postgres.go [hunk 1]
```diff
@@ -505,10 +505,8 @@ func (db *IndexerDb) yieldTxnsThread(ctx context.Context, rows *sql.Rows, result
 			var row idb.TxnRow
 			row.Error = err
 			results <- row
-			rows.Close()
```
Comment (brianolson) [inline idb/postgres/postgres.go:508]: I think these should have become a `defer rows.Close()`

**Comment by fabrice102** [comment]
I've checked using sandbox and the following configurations: (`config.dev` file)

* Develop branch
    ```bash
  export ALGOD_CHANNEL="stable"
  export ALGOD_URL=""
  export ALGOD_BRANCH=""
  export ALGOD_SHA=""
  export NETWORK=""
  export NETWORK_TEMPLATE="/tmp/images/algod/future_template.json"
  export NETWORK_BOOTSTRAP_URL=""
  export NETWORK_GENESIS_FILE=""
  export INDEXER_URL="https://github.com/algorand/indexer"
  export INDEXER_BRANCH="develop"
  export INDEXER_SHA=""
  export INDEXER_DISABLED=""
    ```
* This PR
    ````bash
    export ALGOD_CHANNEL="stable"
    export ALGOD_URL=""
    export ALGOD_BRANCH=""
    export ALGOD_SHA=""
    export NETWORK=""
    export NETWORK_TEMPLATE="/tmp/images/algod/future_template.json"
    export NETWORK_BOOTSTRAP_URL=""
    export NETWORK_GENESIS_FILE=""
    export INDEXER_URL="https://github.com/algorand/indexer"
    export INDEXER_BRANCH="tolik/leaks"
    export INDEXER_SHA=""
    export INDEXER_DISABLED=""
    ```

(do not forget to run
```bash
source config.dev
docker-compose build --no-cache
```
to get the correct versions
)

Then:
```bash
./sandbox goal clerk send -f EQQAHMXWERQXN6QSC3DLFIPW6GTDYKTHNBH7R3ZJVG6FRFVBFVJ4UNGIHI --to EQQAHMXWERQXN6QSC3DLFIPW6GTDYKTHNBH7R3ZJVG6FRFVBFVJ4UNGIHI --amount 10
for i in $(seq 200)
do
  curl "localhost:8980/v2/accounts/EQQAHMXWERQXN6QSC3DLFIPW6GTDYKTHNBH7R3ZJVG6FRFVBFVJ4UNGIHI?round=36"
done
```

With develop:
```
...
{"account":{"address":"EQQAHMXWERQXN6QSC3DLFIPW6GTDYKTHNBH7R3ZJVG6FRFVBFVJ4UNGIHI","amount":4000000000000000,"amount-without-pending-rewards":4000000000000000,"created-at-round":0,"deleted":false,"participation":{"selection-participation-key":"R6s43pKJozfqN/0cOMd1gQ2N4utXBq8L1GI5+CHegsA=","vote-first-valid":0,"vote-key-dilution":10000,"vote-last-valid":3000000,"vote-participation-key":"PJN+4j1Rn6x+MDC2S4K2I9QoUVcn5ua/hGy0MloyVgA="},"pending-rewards":0,"reward-base":0,"rewards":0,"round":36,"sig-type":"sig","status":"Online"},"current-round":37}
...
{"message":"Indexer health error: %s"}
...
curl: (52) Empty reply from server
curl: (52) Empty reply from server
curl: (7) Failed to connect to localhost port 8980: Connection refused
...
```
indicating that everything completely crashes

With this PR: no issue.

Not tested in production though.

File: idb/postgres/postgres.go (hunk 1 above)
Comment (winder) [reply idb/postgres/postgres.go:508]: +1

`rows.Close()` isn't called when this function is invoked by `YieldTxns`

**Review by winder — CHANGES_REQUESTED** [review]
1 small change needed

File: idb/postgres/postgres.go (hunk 1 above)
Comment (tolikzinovyev) [reply idb/postgres/postgres.go:508]: done

**Review by winder — APPROVED** [review]

**Pull request merged**